statsmodels/miscmodels/robust_genericmle.py

```diff
@@ -0,0 +1,418 @@
+        if start_scale is None:
+            # use MAD for now, TODO: use same as in iteration
+            scale = rscale.mad(resid, center=0)
+        else:
+            scale = start_scale
+
+        for i in range(maxiter):
+            # update location
+            weights = self.norm_loc.weights(resid / scale)
+            if self.weights_prior is not None:
+                weights *= self.weights_prior
+            res_loc = self.fit_loc(self.endog, self.exog, weights=weights).fit()
+            params_old = params
+            params = res_loc.params
+            resid = res_loc.resid
+            resid_scaled = resid / scale
+
+            if update_scale:
+                # update scale
+                # Note: use old residuals or new ?
+                weights_scale = self.meef_scale(resid_scaled) / resid_scaled**2
+                #scale_old = scale
+                scale2 = (weights_scale * resid**2).mean()
+                #scale *= scale_old**2
+                scale2 /= self.scale_bias
+                scale = np.sqrt(scale2)
+
+            #check convergence
+            if np.allclose(params, params_old, rtol=1e-6, atol=1e-10):
+                converged = True
+                break
+        else:
+            converged = False
+
+        # Force estimated scale for WLS results
+        res_loc._cache['scale'] = scale**2
+
+        # TODO return Results
+        return res_loc, scale, converged
+
+
+class _RobustGMM(gmm.GMM):
+    """Experimental class for robust M-estimation using estimating equations
+
+    API will change, needs computational improvements
+
+    The residuals are transformed using the norm.psi function. The estimating
+    equation for the scale needs to be provided in a callable meef_scale.
+
+    The scale parameter is exp transformed to keep it positive.
+
+    This uses an instrumental variable definition of the moment conditions.
+    `instruments` are used in the moment conditions and can be set to exog.
+
+    """
+
+    def __init__(self, endog, exog, instrument, norm=None, meef_scale=None):
+        super(_RobustGMM, self).__init__(endog, exog, instrument)
+
+        if norm is None:
+            norm = rnorms.HuberT()
+            #norm = rnorms.TukeyBiweight(1)
+        if meef_scale is None:
+            #stats.norm.expect(lambda x, *args: rnorms.TukeyBiweight(1).rho(x))
+            scale_bias = 0.10903642477287441
+            # TODO: still the bug in TukeyBiweight  + 0.16666667 in this branch
+            meef_scale = lambda x: rnorms.TukeyBiweight(1).rho(x)-scale_bias + 0.16666667
+
+        self.norm = norm
+        self.meef_scale = meef_scale
+
+
+    def momcond(self, params):
+        #params_loc = params[:-1]
+        scale = np.exp(params[-1])
+        fitted_values = self.predict(params, self.exog)
+        resid_pearson = (self.endog - fitted_values) / scale
+
+        mom_loc = self.norm.psi(resid_pearson)[:,None] * self.instrument
```

Comment:
Is this the only definition for use of instruments?  reference Ronchetti?
`psi(resid) * instruments`
versus
`psi(resid * instruments)`
?

check derivative for linear model, if instruments = exog  (no endogeneity)